### PR TITLE
Use overlays to create multiple envs

### LINKED ISSUE #139: Create variants for production and local development
## Description

Since we're using [kustomize](https://github.com/kubernetes-sigs/kustomize/) to simplify the usage of this repo, we could take advtange of the [overlay](https://kubectl.docs.kubernetes.io/references/kustomize/glossary/#overlay) feature to create [variants](https://github.com/kubernetes-sigs/kustomize/#2-create-variants-using-overlays) for different scenarios.

My proposal is to create 2 variants:

- development: for local environments with lower resource requirements and not tied to any cloud offering like AWS load balancers.
- production: equivalent to the current repository tuned for EKS deployments

```
~/wazuh-kubernetes
├── base
│   ├── storage-class.yaml
│   ├── kustomization.yaml
│   └── wazuh-ns.yaml
└── overlays
    ├── development
    │   ├── cpu_count.yaml
    │   └── kustomization.yaml
    └── production
        ├── cpu_count.yaml
        └── kustomization.yaml
```

### Changes for local development (microk8s)

- Change provisioner on storage-class to the one available on microk8s `provisioner: microk8s.io/hostpath`
- Lower resource requests to values within the cluster limits. cpu: `200m`, memory: `512M`, storage: `200Mi`

## Deploying variants

To deploy a variant first it must be generated like this: `kustomize build wazuh-kubernetes/overlays/development`.

### PR DESCRIPTION
**Pull request by xr09**
### Changes

- Created `EKS` and `local-env` with different resource requirements and storage classes.
- Renamed storage class to generic name `wazuh-storage`

This PR closes #139

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,56 +13,72 @@ The *instructions.md* file describes how to deploy Wazuh on Kubernetes.
...
 ## Branches
 
 * `master` branch contains the latest code, be aware of possible bugs on this branch.
-* `local-environment` branch contains modifications for deploying on local environments.
+
 
 ## Local development
 
-To deploy a cluster on your local environment (like Minikube or Kind) use the branch [local-environment](https://github.com/wazuh/wazuh-kubernetes/tree/local-environment/minikube)
+To deploy a cluster on your local environment (like Minikube, Kind or Microk8s) read the instructions on "local-environment.md".
```
Comment (Zenidd) [inline README.md:81]: Hey @xr09, could you link `local-environment.md` to its source file?

File: instructions.md [hunk 2]
```diff
@@ -135,10 +135,15 @@ The required certificates are imported via secretGenerator on the `kustomization
 
 ### Step 3.2: Apply all manifests using kustomize
 
-By using the kustomization.yml we can now deploy the whole cluster in a single command.
+We are using the overlay feature of kustomize two create two variants: `eks` and `local-env`, in this guide we're using `eks`. (For a deployment on a local environment check the guide on `local-environment.md`)
```
Comment (Zenidd) [inline instructions.md:138]: Same here, could you link `local-environment.md` to its source file?

**Comment by Zenidd** [comment]
Hey @xr09, I think it could be a good idea to also link the `instructions.md` file while mentioning it on the `README.md`.

`The instructions.md file describes how to deploy Wazuh on Kubernetes.`

File: README.md (hunk 1 above)
Comment (xr09) [reply README.md]: Fixed on 92edc7e

File: instructions.md (hunk 2 above)
Comment (xr09) [reply instructions.md]: Fixed on 92edc7e

**Review by Zenidd — APPROVED** [review]
LGTM!

**Pull request merged**